### PR TITLE
Email testing

### PR DESCRIPTION
**Pull request by j-jasik**
This PR fixes end of session emails. Admins and Profs get emails specific to their roles, and professors only see their courses and courses they are viewers for.

### CODE REVIEW

File: sciencelabs/email_tab/email_controller.py [hunk 1]
```diff
@@ -85,23 +45,26 @@ def close_session_email(self, session_id):
...
                 session_students = self.session.get_prof_session_students(session_id, professor_course_ids)  # Gets students specific to prof
                 for student in session_students:
                     students_and_courses[student] = self.session.get_report_student_session_courses(session_id, student.id)  # Same as above
-            # send an email
-            self.send_message(subject, render_template('sessions/email.html', **locals()), recipient.email, None, True)
-
+            attendance = 0
+            for course, info in courses_and_info.items():
+                for student in info['students']:
+                    attendance = attendance + 1
```
Comment (c-schwarze) [inline sciencelabs/email_tab/email_controller.py]: Just want to double check that you aren't double counting students. If a student attends 2 times, we should only count them once. *there are exceptions...hehe

File: sciencelabs/email_tab/email_controller.py [hunk 2]
```diff
@@ -85,23 +45,26 @@ def close_session_email(self, session_id):
...
+            if attendance > 0 or 'Administrator' in recipient_role_names:
+                # send an email
+                self.send_message(subject, render_template('sessions/email.html', **locals()), recipient.email, None, True)
 
     def send_message(self, subject, body, recipients, bcc, html=False):
         if app.config['ENVIRON'] != 'prod':
             print('Would have sent email to: ' + str(recipients))
+            subject = subject + recipients
```
Comment (c-schwarze) [inline sciencelabs/email_tab/email_controller.py]: is this for your tests?

File: sciencelabs/db_repository/course_functions.py [hunk 3]
```diff
@@ -70,6 +70,12 @@ def get_professor_courses(self, prof_id):
         return db_session.query(Course_Table)\
             .filter(CourseProfessors_Table.course_id == Course_Table.id)\
             .filter(CourseProfessors_Table.professor_id == prof_id) \
+            .all()
+
+    def get_current_professor_courses(self, prof_id):
```
[the remark this answers is not in the document]
Comment (j-jasik) [reply sciencelabs/db_repository/course_functions.py]: It's not done here, but it is done in the email_controller.

File: sciencelabs/db_repository/user_functions.py [hunk 4]
```diff
@@ -426,9 +426,10 @@ def get_bcc_emails(self, bcc_ids):
         return emails
 
     def get_end_of_session_recipients(self):
-        # todo: ideally we wouldn't use id's, we would use the name.
-        admins = self.get_users_in_group(40001)  # Id for admins
-        profs = self.get_users_in_group(40005)  # Id for profs
+        admin_role = self.get_role_by_name('Administrator')
+        prof_role = self.get_role_by_name('Professor')
+        admins = self.get_users_in_group(admin_role.id)
+        profs = self.get_users_in_group(prof_role.id)
```
[the remark this answers is not in the document]
Comment (j-jasik) [reply sciencelabs/db_repository/user_functions.py]: 😄

File: sciencelabs/email_tab/email_controller.py (hunk 1 above)
Comment (j-jasik) [reply sciencelabs/email_tab/email_controller.py]: That is checked for, but not here. This is just making sure that a professor only receives an email if they have some students who attended the session.

File: sciencelabs/email_tab/email_controller.py (hunk 2 above)
Comment (j-jasik) [reply sciencelabs/email_tab/email_controller.py]: Yes, but also if the environment is not prod, we don't want to actually send emails.

Comment (c-schwarze) [reply sciencelabs/email_tab/email_controller.py]: ah, didn't see that.

File: sciencelabs/email_tab/email_controller.py (hunk 1 above)
Comment (c-schwarze) [reply sciencelabs/email_tab/email_controller.py]: 👍

File: sciencelabs/db_repository/course_functions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (c-schwarze) [reply sciencelabs/db_repository/course_functions.py]: 👍

**Pull request merged**